listings/ch00-introduction/struct_as_mapping_key/src/lib.cairo

```diff
@@ -0,0 +1,26 @@
+#[contract]
+mod StructAsMappingKey {
+    use starknet::get_caller_address;
+    use starknet::ContractAddress;
+    use poseidon::poseidon_hash_span;
+
+    struct Storage {
+        _map: LegacyMap::<ContractAddress, felt252>, 
+    }
+
+    struct User {
+        name : felt252,
+        age : felt252,
+        address : ContractAddress,
+    }
+
+    #[external]
+    fn set(key: User, value: felt252) {
+        _map::write(poseidon_hash_span(key.span()), value);
```

Comment:
It does not work as it expects a span, but span is only for arrays and key is not an array (it's a struct), so I guess we have to use another function

Reply:
your struct needs to derive `Serde` and then you can call `Serde::<User>::serialize(key,serialized_struct)` to make it an array (check corelib to see the correct syntax to use `Serde`